### PR TITLE
SDK Update, range fix, some small changes

### PR DESCRIPTION
**Pull request by someburner**
PR addresses the following:

 - Update SDK submodule to latest
 - range args fix (uses a newer SDK macro) 
 - compiler warnings for const
 - enumerate agg types, helps get rid of some magic numbers that add to confusion with argc numbers

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -100,30 +100,44 @@ void ReplyWithAggValue(RedisModuleCtx *ctx, timestamp_t last_agg_timestamp, Aggr
...
     long long start_ts, end_ts;
+    long long time_delta = 0;
+    RedisModuleString * aggTypeStr = NULL;
+
+    int pRes = REDISMODULE_ERR;
+    switch (argc) {
+        case 4:
+            pRes = RMUtil_ParseArgs(argv, argc, 2, "ll", &start_ts, &end_ts);
+            break;
+        /* case 5: return RedisModule_ReplyWithLongLong(ctx, 12341234); */ // Test false positive
```
Comment (danni-m) [inline src/module.c]: @someburner i think this line can be safely deleted, any reason you kept this?

File: src/module.c [hunk 2]
```diff
@@ -100,30 +100,44 @@ void ReplyWithAggValue(RedisModuleCtx *ctx, timestamp_t last_agg_timestamp, Aggr
...
-        int aggType = StringAggTypeToEnum(argv[4]);
-        if (aggType < 0 && aggType > 5) {
+    AggregationClass *aggObject = NULL;
+
+    if (argc > 4)
+    {
+        if (!aggTypeStr)
+            return RedisModule_ReplyWithError(ctx, "TSDB: Unkown aggregation type");
+        agg_type = StringAggTypeToEnum( aggTypeStr );
+        // int aggType = StringAggTypeToEnum( aggTypeStr );
```
Comment (danni-m) [inline src/module.c]: @someburner i think this line can be safely deleted, any reason you kept this?

**Comment by someburner** [comment]
@danni-m 

For `SeriesMemUsage`, the issue is that in the SDK.h header we have:

```c
typedef struct RedisModuleTypeMethods {
    uint64_t version;
    RedisModuleTypeLoadFunc rdb_load;
    RedisModuleTypeSaveFunc rdb_save;
    RedisModuleTypeRewriteFunc aof_rewrite;
    RedisModuleTypeMemUsageFunc mem_usage;
    RedisModuleTypeDigestFunc digest;
    RedisModuleTypeFreeFunc free;
} RedisModuleTypeMethods;

typedef size_t (*RedisModuleTypeMemUsageFunc)(const void *value);
```

But actually I don't know that part of the SDK at all, so perhaps that is not OK?

File: src/module.c (hunk 2 above)
Comment (someburner) [reply src/module.c]: I just wasn't sure why there was 2 calls to `StringAggTypeToEnum` in the first place, so wanted to leave to make sure there isn't something I was missing. So if not, then yes we can delete.

File: src/module.c (hunk 1 above)
Comment (someburner) [reply src/module.c]: It was to test my test-cases (since I was testing for crash by loading the module, not using the py tests). So yeah can be removed.

**Comment by danni-m** [comment]
@someburner oh, makes sense. if the function typedef contains const we should have our function with const as well.

**Pull request merged**